cli_client/python/timesketch_cli_client/cli.py

```diff
@@ -0,0 +1,146 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Timesketch CLI client."""
+
+import sys
+import click
+
+from requests.exceptions import ConnectionError as RequestConnectionError
+
+from timesketch_api_client import config as timesketch_config
+
+from timesketch_cli_client.commands import analyze
+from timesketch_cli_client.commands import config
+from timesketch_cli_client.commands import importer
+from timesketch_cli_client.commands import search
+from timesketch_cli_client.commands import sketch as sketch_command
+from timesketch_cli_client.commands import timelines
+
+from .definitions import DEFAULT_OUTPUT_FORMAT
+from .version import get_version
+
+
+class TimesketchCli(object):
+    """Timesketch CLI state object.
+
+    Attributes:
+        sketch_from_flag: Sketch ID if provided by flag
+        config_assistant: Instance of ConfigAssistant
+    """
+    def __init__(self, api_client=None, sketch_from_flag=None, conf_file=None):
+        """Initialize the state object.
+
+        Args:
+            sketch_from_flag: Sketch ID if provided by flag.
+        """
+        self.api = api_client
+        self.sketch_from_flag = sketch_from_flag
+
+        if not api_client:
+            try:
+                self.api = timesketch_config.get_client(load_cli_config=True)
+                if not self.api:
+                    raise RequestConnectionError
+            except RequestConnectionError:
+                click.echo('ERROR: Cannot connect to the Timesketch server.')
+                sys.exit(1)
+
+        self.config_assistant = timesketch_config.ConfigAssistant()
+        if conf_file:
+            self.config_assistant.load_config_file(
+                conf_file, load_cli_config=True)
+        else:
+            self.config_assistant.load_config_file(load_cli_config=True)
+
+    @property
+    def sketch(self):
+        """Sketch object from the API client.
+
+        Returns:
+            Sketch object.
+        """
+        active_sketch = None
+        sketch_from_config = self.config_assistant.get_config('sketch')
+
+        if self.sketch_from_flag:
+            active_sketch = self.api.get_sketch(
+                sketch_id=int(self.sketch_from_flag))
+        elif sketch_from_config:
+            active_sketch = self.api.get_sketch(
+                sketch_id=int(sketch_from_config))
+
+        if not active_sketch:
+            click.echo(('ERROR: You need to specify a sketch, either with a '
+                        'flag (--sketch <SKETCH ID>) or update the config.'))
+            sys.exit(1)
+
+        # Make sure we have access to the sketch.
+        try:
+            active_sketch.name
+        except KeyError:
+            click.echo('ERROR: No such sketch or you don\'t have access.')
+            sys.exit(1)
+
+        return active_sketch
+
+    @property
+    def output_format(self):
+        """Get or set the default output format.
```

Comment:
well, this is not a set operation, this is a property, so it is only a get.... if you need a setter, then yo uneed to define a 

```python
@output_format.setter
def output_format(self, output_format)
```

function

Reply:
Ah, good catch. This is not a setter, but it does set the format to the default value if missing from the active config. Changed the wording.